### DISCUSSION TITLE
"Visited" Map error - db query error: pq: the database system is in recovery mode

### DISCUSSION THREAD

**Discussion by burratha** [teslamate-org/teslamate#3265] [Q&A]
Hi

I'm having trouble viewing the "Visited" map on my Teslamate install (Raspberry pi / Docker). The map will attempt to load, however I will get the following error, "db query error: pq: the database system is in recovery mode"

I can manually change the date parameters to "last 30 days" but anything older than that causes an error. All the relevant data appears as normal when searching through the other dashboards - it's just the overview map where the problem seems to lie.

Can anyone help, please?

Thanks

**Comment by lvanjaarsveld** [comment]
Anyone? Same issue here.

**Comment by Ruud-C** [comment]
Exactly the same problem here after I upgraded TeslaMate to the latest version (Docker install).

**Comment by sparvata** [comment]
from your docker compose directory execute the command below and see if there are any errors from database
docker-compose logs database

**Reply by Ophrys78** [reply]
I have the exact same issue which generates the following database log:

_database_1 | 2023-09-04 20:20:35.249 UTC [1] LOG: server process (PID 8342) was terminated by signal 11: Segmentation fault
database_1 | 2023-09-04 20:20:35.249 UTC [1] DETAIL: Failed process was running: SELECT
database_1 | to_timestamp(floor((extract('epoch' FROM date) / 60)) * 60) AT TIME ZONE 'UTC' AS time,
database_1 | avg(longitude) AS lng
database_1 | FROM
database_1 | positions
database_1 | WHERE
database_1 | car_id = '1' AND
database_1 | date BETWEEN '2018-09-04T20:20:22.571Z' AND '2023-09-04T20:20:22.572Z'
database_1 | GROUP BY
database_1 | 1
database_1 | ORDER BY
database_1 | 1 ASC
database_1 | 2023-09-04 20:20:35.249 UTC [1] LOG: terminating any other active server processes
database_1 | 2023-09-04 20:20:35.371 UTC [1] LOG: all server processes terminated; reinitializing
database_1 | 2023-09-04 20:20:37.209 UTC [8347] LOG: database system was interrupted; last known up at 2023-09-04 20:19:46 UTC
database_1 | 2023-09-04 20:20:37.215 UTC [8350] FATAL: the database system is in recovery mode
database_1 | 2023-09-04 20:20:37.218 UTC [8351] LOG: PID 8331 in cancel request did not match any process
database_1 | 2023-09-04 20:20:37.233 UTC [8354] LOG: PID 8337 in cancel request did not match any process
database_1 | 2023-09-04 20:20:37.234 UTC [8353] LOG: PID 8339 in cancel request did not match any process
database_1 | 2023-09-04 20:20:37.238 UTC [8352] LOG: PID 8332 in cancel request did not match any process
database_1 | 2023-09-04 20:20:37.250 UTC [8356] LOG: PID 8334 in cancel request did not match any process
database_1 | 2023-09-04 20:20:37.258 UTC [8357] LOG: PID 8335 in cancel request did not match any process
database_1 | 2023-09-04 20:20:37.261 UTC [8358] LOG: PID 8338 in cancel request did not match any process
database_1 | 2023-09-04 20:20:37.262 UTC [8355] LOG: PID 8340 in cancel request did not match any process
database_1 | 2023-09-04 20:20:37.266 UTC [8359] LOG: PID 8336 in cancel request did not match any process
database_1 | 2023-09-04 20:20:37.274 UTC [8360] LOG: PID 8333 in cancel request did not match any process
database_1 | 2023-09-04 20:20:37.279 UTC [8361] FATAL: the database system is in recovery mode
database_1 | 2023-09-04 20:20:37.293 UTC [8363] FATAL: the database system is in recovery mode
database_1 | 2023-09-04 20:20:37.304 UTC [8362] FATAL: the database system is in recovery mode
database_1 | 2023-09-04 20:20:37.305 UTC [8364] FATAL: the database system is in recovery mode
database_1 | 2023-09-04 20:20:37.315 UTC [8365] FATAL: the database system is in recovery mode
database_1 | 2023-09-04 20:20:37.324 UTC [8366] FATAL: the database system is in recovery mode
database_1 | 2023-09-04 20:20:37.339 UTC [8367] FATAL: the database system is in recovery mode
database_1 | 2023-09-04 20:20:37.345 UTC [8368] FATAL: the database system is in recovery mode
database_1 | 2023-09-04 20:20:37.348 UTC [8369] FATAL: the database system is in recovery mode
database_1 | 2023-09-04 20:20:37.365 UTC [8370] FATAL: the database system is in recovery mode
database_1 | 2023-09-04 20:20:37.438 UTC [8347] LOG: database system was not properly shut down; automatic recovery in progress
database_1 | 2023-09-04 20:20:37.459 UTC [8347] LOG: redo starts at 0/280EB9D0
database_1 | 2023-09-04 20:20:37.459 UTC [8347] LOG: invalid record length at 0/280EBA08: wanted 24, got 0
database_1 | 2023-09-04 20:20:37.460 UTC [8347] LOG: redo done at 0/280EB9D0 system usage: CPU: user: 0.00 s, system: 0.00 s, elapsed: 0.00 s
database_1 | 2023-09-04 20:20:37.490 UTC [8348] LOG: checkpoint starting: end-of-recovery immediate wait
database_1 | 2023-09-04 20:20:37.552 UTC [8348] LOG: checkpoint complete: wrote 3 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.017 s, sync=0.010 s, total=0.075 s; sync files=2, longest=0.007 s, average=0.005 s; distance=0 kB, estimate=0 kB
database_1 | 2023-09-04 20:20:37.580 UTC [1] LOG: database system is ready to accept connections_

**Comment by lvanjaarsveld** [comment]
Identical as above.

**Comment by alexeiw123** [comment]
I appear to be having a very similar issue.
My database holds about 3 years of data and I have nightly backups going back 7 days.

Last week I did a `docker compose pull` then on restart my teslamate container kept restarting. Through inspection of the logs it looks like my database is going in to recovery mode. What has me stumped is that I am not able to restore any of my previous 7 days backups. I can deploy a fresh installation of teslamate in either the current or previous version which will appear stable and when I restore my backup, it gets a long way through it before having an error:

```
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
connection to server was lost
lex@teslamate:~ $
```

After the 'failed' backup, another curious observation is that my drive data appears to be in-tact in the database, only that it keeps entering recovery mode if the teslamate container is running. I can do a `docker compose stop teslamate` and then I am able to interact with the database using grafana just fine.

Would love to find a solution to this.
